src/base/utils/string.h

```diff
@@ -39,15 +42,47 @@ namespace Utils
 {
     namespace String
     {
+        class NaturalCompare
+        {
+        public:
+            explicit NaturalCompare(const Qt::CaseSensitivity caseSensitivity = Qt::CaseSensitive);
+            int operator()(const QString &left, const QString &right) const;
+
+        private:
+            int compare(const QString &left, const QString &right) const;
+
+            const Qt::CaseSensitivity m_caseSensitivity;
+#ifndef Q_OS_WIN
```

Comment:
both private method and const above are used in alternative branch only, isn't it?

Reply:
yes, how about leaving the const variable out of if conditions? wrapping in #if looks weird.

Reply:
>wrapping in #if looks weird.

Why?
Why we need to have unused class member?

Reply:
@Chocobo1, I'm sorry I tortured you... I think now I have idea how to minimize private code in `string.h`. We can have only `naturalCompare(QString, QString, Qt::CaseSensitivity)` here (and naturalLessThan template). Do you interested in it?

Reply:
>Why?

having a #if for class variable initialization looks weird for me, I'm not used to it.
Anyway I've changed it as requested.

>Do you interested in it?

I think it's better to let you handle it in another PR. :)

Reply:
I'm sorry one more time, but this part of changes was caused by my suggestion "to merge two naturalCompareXXX functions into one" and now it leads to even more bad code in header file. So please revert it to your previous form with `naturalCompare(QString, QString, Qt::CaseSensitivity)` and two static variables in it (or even to two naturalCompareXXX functions, if you want). Then I implement my changes in another PR.

Reply:
>So please revert it to your previous form 

Good thing I made a backup branch before experimenting, otherwise I'll might just let this rot.